operate/services/manage.py

```diff
@@ -546,12 +557,66 @@ def _deploy_service_onchain_from_safe(  # pylint: disable=too-many-statements,to
                 service_registry_token_utility="0xa45E64d13A30a51b91ae0eb182e88a40e9b18eD8",  # nosec
                 min_staking_deposit=20000000000000000000,
                 activity_checker=NULL_ADDRESS,  # nosec
-                agent_mech="0x77af31De935740567Cf4fF1986D04B2c964A786a",  # nosec
             )
 
         # TODO A customized, arbitrary computation mechanism should be devised.
         env_var_to_value = {}
         if chain == service.home_chain:
+            # Try if activity checker is a MechActivityChecker contract
+            try:
+                mech_activity_contract = t.cast(
+                    MechActivityContract,
+                    MechActivityContract.from_dir(
+                        directory=str(DATA_DIR / "contracts" / "mech_activity")
+                    ),
+                )
+
+                agent_mech = (
+                    mech_activity_contract.get_instance(
+                        ledger_api=sftxb.ledger_api,
+                        contract_address=staking_params["activity_checker"],
+                    )
+                    .functions.agentMech()
```

Comment:
So, here we assume that an activity checker contract will never have both the `agentMech` and `mechMarketplace` methods, right?

Reply:
Yes. Activity checkers are subclassed from a base class. In the case of the variables set for the Trader, these 2 cases are determined based on a function call that exists on each contract, but not on the other (that's why catching the exception is required).

Possibly can be done in a nicer way when we clean up the code.